main.go

```diff
@@ -54,27 +61,38 @@ func actionMain(_ map[string]string, _ *ezactions.RunResources) (map[string]stri
 	oauthClient := oauth2.NewClient(ctx, tokenSource)
 	githubClient := github.NewClient(oauthClient)
 
-	for {
-		response, err := createFile(eventPullRequestTitle, eventPullRequestIssuer, pullRequestNumber, githubClient, ctx)
-		if response != nil && response.StatusCode == 422 {
-			log.Printf("file already exists for %v: %v, considering as success", pullRequestNumber, err)
-			return map[string]string{}, nil
-		}
-		if err != nil {
-			if _, ok := err.(*github.RateLimitError); ok {
-				waitDuration := time.Now().Sub(response.Rate.Reset.Time)
-				log.Printf("hit rate limit, will need to wait %v sec and retry", waitDuration.Seconds())
-				time.Sleep(waitDuration)
-			} else {
-				return nil, fmt.Errorf("failed to create file for %v: %v", pullRequestNumber, err)
-			}
+	pullRequestTitle := eventPullRequestTitle
+	pullRequestIssuer := eventPullRequestIssuer
+
+	if len(pullRequestTitle) == 0 {
+		pr, _, err := githubClient.PullRequests.Get(ctx, "apiiro", "lim", pullRequestIntNumber)
+		if err == nil {
+			pullRequestTitle = pr.GetUser().GetLogin()
+			pullRequestIssuer = pr.GetTitle()
+		} else {
+			log.Printf("failed to find pr title and issuer for %v: %v", pullRequestNumber, err)
 		}
-		if response.StatusCode == 201 {
-			log.Printf("completed successfully!")
-			return map[string]string{}, nil
+	}
+
+	response, err := createFile(pullRequestTitle, pullRequestIssuer, pullRequestNumber, githubClient, ctx)
+	if response != nil && response.StatusCode == 422 {
+		log.Printf("file already exists for %v: %v, considering as success", pullRequestNumber, err)
+		return map[string]string{}, nil
+	}
+	if err != nil {
+		if _, ok := err.(*github.RateLimitError); ok {
+			waitDuration := time.Since(response.Rate.Reset.Time)
+			log.Printf("hit rate limit, will need to wait %v sec and retry", waitDuration.Seconds())
+			time.Sleep(waitDuration)
```

Comment:
I think the `for` somehow (or at least intended to be) used by this line as a retry mechanism, otherwise, I don't understand this `.Sleep` statement 😅

Reply:
@barapiiro  Yeah maybe, i thought that also. 
What do you suggest i do? I see 3 options:
1. not touch it
2. remove the sleep
3. return the for but remove the 'return' at the end of the for loop, so the retry mechanism would (potentially) work.

Reply:
I really don't know how `for` loops works in `go`... 
from reading the code before your changes, maybe it simply did nothing, or maybe it somehow magically knows that you returned a `nil` and then simply retries it 🤷🏽 
if you know how to fix the retry mechanism, I think it's better (+ limit it to 3 retries and not "forever")

Reply:
There's no magic there... for loops works as you would expect them to work, the retry just didn't work previously. The go compiler had a warning that this for loop will only be executed once.
I think i'll just keep it as-is for now, if we see we have a rate limit error then we can go back and fix it with a tested retry mechanism